### PR TITLE
refactor(motion_utils): delete wall marker explicitly 

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

Originallyn , wall markers were displayed at stop/slow line and is dispearred after a certain period of time. 
With this change, if the module's processing determines that there is no stop/slow line, the existing wall marker will be deleted.

## Pre-review checklist for the PR author
The PR author must check the checkboxes below when creating the PR.

 I've confirmed the [contribution guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/).
 The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
## In-review checklist for the PR reviewers

- [ ]  The PR reviewers must check the checkboxes below before approval.
- [ ] The PR follows the [pull request guidelines](https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/).
## Post-review checklist for the PR author
The PR author must check the checkboxes below before merging.
- [ ] There are no open discussions or they are tracked via tickets.
After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp [hunk 1]
```diff
@@ -31,7 +31,8 @@ StopLineModule::StopLineModule(
 : SceneModuleInterface(module_id, logger, clock),
   lane_id_(lane_id),
   stop_line_(stop_line),
-  state_(State::APPROACH)
+  state_(State::APPROACH),
+  last_state_(State::START)
```
Comment (TakaHoribe) [inline planning/behavior_velocity_planner/src/scene_module/stop_line/scene.cpp:35]: `last_state_` is not used anywhere.

**Comment by TakaHoribe** [comment]
@kyoichi-sugahara Please write description

File: common/motion_utils/include/motion_utils/marker/marker_helper.hpp [hunk 2]
```diff
@@ -41,6 +46,46 @@ visualization_msgs::msg::MarkerArray createDeletedSlowDownVirtualWallMarker(
...
+  const std::vector<Pose> & stop_poses, const std::vector<Pose> & previous_poses,
+  const rclcpp::Time & now, int32_t id);
+
+using create_wall_function = std::function<visualization_msgs::msg::MarkerArray(
+  const geometry_msgs::msg::Pose & pose, const std::string & module_name, const rclcpp::Time & now,
+  const int32_t id, const double longitudinal_offset)>;
+
+using delete_wall_function =
+  std::function<visualization_msgs::msg::MarkerArray(const rclcpp::Time & now, const int32_t id)>;
+
```
Comment (soblin) [inline common/motion_utils/include/motion_utils/marker/marker_helper.hpp:60]: Can you define `create_wall_function` and `delete_wall_function` in `VirtualWallMarkerCreator`:
```
class VirtualWallMarkerCreator {
   using create_wall_function = ...
   using delete_wall_function -= ...
}
```
to limit the scope of symbol ?

Comment (kyoichi-sugahara) [reply common/motion_utils/include/motion_utils/marker/marker_helper.hpp]: Thanks!

**Review by soblin — APPROVED** [review]
LGTM

File: common/motion_utils/src/marker/marker_helper.cpp [hunk 3]
```diff
@@ -128,4 +131,77 @@ visualization_msgs::msg::MarkerArray createDeletedDeadLineVirtualWallMarker(
...
+visualization_msgs::msg::MarkerArray VirtualWallMarkerCreator::createStopVirtualWallMarker(
+  const std::vector<Pose> & stop_poses, const std::string & module_name, const rclcpp::Time & now,
+  int32_t id, const double longitudinal_offset)
+{
+  create_wall_function f1 = motion_utils::createStopVirtualWallMarker;
+  delete_wall_function f2 = motion_utils::createDeletedStopVirtualWallMarker;
+
+  return handleVirtualWallMarker(
+    stop_poses, module_name, now, id, f1, f2, previous_stop_poses_, longitudinal_offset);
+}
```
[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply common/motion_utils/src/marker/marker_helper.cpp:183]: I applied your proposal thanks!

**Pull request merged**